api/client/src/main/java/org/projectnessie/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -201,7 +201,7 @@ private Tokens fetchNewTokens(String code) {
             .clientId(params.getClientId())
             .scope(params.getScope().orElse(null))
             .build();
-    HttpRequest request = httpClient.newRequest().path(params.getTokenEndpoint().getPath());
+    HttpRequest request = httpClient.newRequest().path(params.getResolvedTokenEndpoint().getPath());
```

Comment:
I'm a bit uneasy about reusing URI paths on top of the base URI, especially with discovery supported. OpenID specs do not require endpoints to have the same host and they explicitly [allow](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata) using different ports for different endpoints.

Would it be too much trouble to use full endpoint URIs for requests? It might require a parent PR for refactoring, I guess... but ultimately it looks like the "base URI" concept exists only in the Nessie Client, HTTP client implementations can send different requests to different servers. WDYT?

Reply:
Yes I agree, but the consequence of doing as you suggested is that we'd need one `HttpClient` per endpoint. But let's try, it's probably more future-proof anyway.

Reply:
I suggest refactoring `HttpClient` to allow independent URIs in each request (as the underlying HTTP client allow)

Reply:
Then in this case I suggest we do it in 2 steps: in this PR I will make it so that there is one `HttpClient` per endpoint, then in a follow-up PR we refactor `HttpClient`. I think this is acceptable because so far there is only one endpoint constantly used, the token one.

Reply:
SGTM :+1: